### PR TITLE
Convert existing unit tests to standard layout

### PR DESCRIPTION
**Pull request by dereknola**
#### Proposed Changes ####
This PR coverts 4 existing unit tests to the standard unit test layout being pushed. External dependencies and "Assert" functions have been removed. Additionally, a new testlet for parser_test.go was added to improve test coverage (and also showing the advantage of adding additional testing using this format).

#### Types of Changes ####
Enhancement 

#### Verification ####
Run `go tests ./pkg/...`

#### Linked Issues ####
https://github.com/k3s-io/k3s/issues/3508
#### User-Facing Change ####
```release-note
NONE
```

### CODE REVIEW

File: pkg/authenticator/hash/scrypt_test.go [hunk 1]
```diff
@@ -3,28 +3,43 @@ package hash
...
-	assert.Nil(t, hasher.VerifyHash(hash, secretKey))
-	assert.NotNil(t, hasher.VerifyHash(hash, secretKey+":wrong!"))
+func TestSCrypt_VerifyHash(t *testing.T) {
+	type args struct {
+		secretKey string
+	}
+	tests := []struct {
+		name     string
+		args     args
+		setup    func() error // Optional, delete if unused
```
Comment (briandowns) [inline pkg/authenticator/hash/scrypt_test.go:15]: Should these be deleted?

**Review by brandond — APPROVED** [review]

File: pkg/authenticator/hash/scrypt_test.go (hunk 1 above)
Comment (briandowns) [reply pkg/authenticator/hash/scrypt_test.go:15]: There is another ref below to these unused functions.

Comment (dereknola) [reply pkg/authenticator/hash/scrypt_test.go:15]: Yes they should!

**Review by briandowns — APPROVED** [review]
LGTM

**Pull request merged**